infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/StreamsSnapshotWriter.java

```diff
@@ -157,7 +157,12 @@ private void processUpdatesShadowStream(List<SMREntry> smrEntries, Long currentS
 
         try (TxnContext txn = metadataManager.getTxnContext()) {
             updateLog(txn, smrEntries, shadowStreamUuid);
-            metadataManager.updateReplicationMetadataField(txn, session, ReplicationMetadata.LASTSNAPSHOTTRANSFERREDSEQNUMBER_FIELD_NUMBER, currentSeqNum);
+            ReplicationMetadata metadata = metadataManager.queryReplicationMetadata(txn, session);
+            ReplicationMetadata updatedMetadata = metadata.toBuilder().setLastSnapshotStarted(srcGlobalSnapshot)
```

Comment:
Just curious - are these changes needed for the uncompressed size handling or is it a separate change?

Reply:
they are indirectly needed..

Earlier, we wrote the updated metadata multiple times in a given transaction, once for each field(topologyId, last snapshot started and last transferred).  This increased the transaction size and was not required as all fields are in the same message.  I have consolidated it in a single update.

Let me know if you have any further questions.